### ISSUE TITLE
[Failing test] ci-golang-tip-k8s-1-18 fails very frequently

### ISSUE THREAD

**Issue by tosi3k** [kubernetes/kubernetes#97846]
**Which jobs are failing**:
ci-golang-tip-k8s-1-18

**Since when has it been failing**:
The job started to fail with high frequency around Dec 9th; first failing job run: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-golang-tip-k8s-1-18/1336941587187372032.

**Testgrid link**:
https://testgrid.k8s.io/sig-scalability-golang#golang-tip-k8s-1-18

**Reason for failure**:
Multiple API call latency SLOs are being violated: GETs of leases, endpoints, deployments, DELETEs of secrets, pods, etc.

**Anything else we need to know**:
This job runs a Clusterloader2 performance test on a cluster using a fixed version of k8s binaries but built using Golang tip commit, hence the performance regression almost surely comes from Golang.

The Golang commit used as the tip in the first failing run is [this](https://go-review.googlesource.com/c/go/+/276652) one which means the regression was introduced no later than therein. Given that this commit introduced changes only in ARM/ARM64 and MIPS/MIPS64 architectures, the regression was already present earlier.

/sig scalability
/sig release

**Comment by tosi3k** [comment]
I will run some experimental jobs to narrow the area in which the regression was introduced.

/assign

**Comment by tosi3k** [comment]
Ran an experiment using [89f465c2b5](https://go-review.googlesource.com/c/go/+/276552) as the tip Golang commit to confirm a regression exists upon this point of time.

I will run an experiment using [5627a4dc30](https://go-review.googlesource.com/c/go/+/276452) as the next tip (this is the k8s build directly preceding 89f465c2b5 that was generated by [build-and-push-k8s-at-golang-tip](https://testgrid.k8s.io/sig-scalability-golang#build-and-push-k8s-at-golang-tip) CI job).

**Comment by dims** [comment]
subscribe

**Comment by justaugustus** [comment]
FYI @kubernetes/release-engineering 
ref: https://github.com/kubernetes/release/issues/1834

**Comment by wojtek-t** [comment]
It seems we're mostly understanding the issue - we're trying to confirm last remaining details.

But the bottomline is that it doesn't seem to be golang-related - golang seem to be fine.

We will update the issue once we confirm the remaining details.

**Comment by justaugustus** [comment]
Thanks for the update, @wojtek-t!

**Comment by wojtek-t** [comment]
To add a bit more details to it - there were some internal changes which resulted in some underlying machine architecture having much more predictible networking characteristics than others.

With our jobs requiring high predictability, the recommendation from Google networking engineers was to simply migrate the jobs to new enough architecture (Skylake+) which we did in: https://github.com/kubernetes/test-infra/pull/20584

And that fixed our problems.

Closing this one as addressed.
